src/test/common/test_mutex.cc

```diff
@@ -63,7 +63,8 @@ TEST(Mutex, RecursiveWithoutLockdep) {
 TEST(Mutex, DeleteLocked) {
   Mutex* m = new Mutex("Recursive3",false);
   m->Lock();
-  unsetprdumpable();
-  EXPECT_DEATH(delete m,".*");
-  setprdumpable();
+  {
```

Comment:
@badone @tchaikov 
Interesting concept. This was on my to do list, since I usually end up with directories littered with *core files..
It is possible to set a nodump flag globally before starting the tests, but then good dumps also get lost...
Under FreeBSD I think the best way to go is set the rlimit on coresize to 0. (after saving the max limit first.) . That would not be a very difficult addition which I'll do once this PR is in.